src/deserialize.ts

```diff
@@ -74,7 +74,85 @@ export const defaultNodeTypes: NodeTypes = {
   image: 'image',
 };
 
-export default function deserialize(node: MdastNode, opts?: OptionType) {
+type TextNode = { text?: string | undefined };
+
+export type DeserializedNode =
+  | {
+      type: 'code_block';
+      language: string | undefined;
+      children: Array<TextNode>;
```

Comment:
FYI, I had to make these `Array<TextNode>` instead of the preferred (but newer) syntax `TextNode[]`. This is because I kept getting blocked by the lint error:

```
/Users/davidgolightly/dev/remark-slate/src/deserialize.ts
  0:0  error  Parsing error: Cannot read property 'map' of undefined
```

This is commonly caused when the TypeScript version of the project and the TypeScript version used by ESLint are wildly out of sync. It looks like the version used by ESLint is provided by [TSDX](https://github.com/jaredpalmer/tsdx) and not checked in to the project. Long term, I'd recommend migrating away from using TSDX since it hasn't been updated in over 15 months at this point and is only likely to fall further behind the latest TS.

Reply:
Thanks, and yeah, I'm not opposed to moving off of tsdx